### ISSUE TITLE
"Non-exhaustive patterns in function hasNum" error

### ISSUE THREAD

**Issue by mafredri** [koalaman/shellcheck#532]
Input:

```shell
#!/bin/sh

# shellcheck source=source/config/settings.sh
. ${VAR}/config/settings.sh
```

Gives:
```
shellcheck: ShellCheck/Analytics.hs:226:9-46: Non-exhaustive patterns in function hasNum
```

On OS X El Capitan, Shellcheck v. 0.4.1

Only seems to be a problem when `$VAR` is unknown. Can also be worked around by wrapping `$VAR` in quotes, like so: `"${VAR}"`.

**Comment by koalaman** [comment]
Entirely right. This was fixed just a few weeks ago in b9cb040.

**Comment by mafredri** [comment]
Ah, thanks. Didn't see a previous issue for it, but glad to see it's fixed. Cheers!
